examples/actions.spec.ts

```diff
@@ -0,0 +1,52 @@
+import test from 'ava';
+import { of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { actionCreator } from 'rxbeach';
+import { extractPayload, ofType } from 'rxbeach/operators';
+
+const voidAction = actionCreator('[test] void action');
+const primitiveAction = actionCreator<number>('[test] primitive action');
+type Payload = { foo: number };
+const payloadAction = actionCreator<Payload>('[test] payload action');
+
+test('creates action objects', t => {
+  const a = voidAction();
+  const b = primitiveAction(321);
+  const c = payloadAction({ foo: 123 });
+
+  t.is((a as any).payload, undefined);
+  t.is(b.payload, 321);
+  t.deepEqual(c.payload, { foo: 123 });
+});
+
+test('can filter actions', async t => {
+  const a = primitiveAction(12);
+  const r = await of(voidAction(), a, payloadAction({ foo: 123 }))
+    .pipe(ofType(primitiveAction))
+    .toPromise();
```

Comment:
I was a bit late to the party here, but I still think we should be using marble tests for testing streams.

Converting streams to a promise is an generally considered an anti-pattern and I think it makes sense to avoid doing that in tests as well.